File: typescript/packages/bench/src/exporters/htmlExporter.ts
/** Minimal table markup. */

import type { ReportRow } from "../contracts/reportContract";
import { AbstractExporter } from "../support/abstractExporter";

/** Minimal table markup. */
export class HtmlExporter extends AbstractExporter {
    constructor() {
        super("html", "text/html");
    }

    override export(rows: readonly ReportRow[]): string {
        const body = rows
            .map((row) => `<tr><td>${row.label}</td><td>${this.formatCents(row.cents)}</td></tr>`)
            .join("");

        return `<table>${body}</table>`;
    }
}

File: rust/crates/atelier-bench/src/exporters/html_exporter.rs
//! Minimal table markup.

use crate::contracts::exporter_contract::ExporterContract;
use crate::contracts::formatter_contract::FormatterContract;
use crate::contracts::report_contract::ReportRow;
use crate::support::abstract_exporter::ExporterBase;

/// Minimal table markup.
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub struct HtmlExporter {
    base: ExporterBase,
}

impl HtmlExporter {
    pub const fn new() -> Self {
        Self {
            base: ExporterBase::new("html", "text/html"),
        }
    }

    /// Shared extension/mime pair and the cent formatter.
    pub const fn base(&self) -> &ExporterBase {
        &self.base
    }
}

impl Default for HtmlExporter {
    fn default() -> Self {
        Self::new()
    }
}

impl ExporterContract for HtmlExporter {
    fn extension(&self) -> &'static str {
        self.base.extension
    }

    fn mime(&self) -> &'static str {
        self.base.mime
    }

    fn export(&self, rows: &[ReportRow]) -> String {
        let body = rows
            .iter()
            .map(|row| {
                format!(
                    "<tr><td>{}</td><td>{}</td></tr>",
                    row.label,
                    self.base.format_cents(row.cents)
                )
            })
            .collect::<Vec<_>>()
            .join("");

        format!("<table>{body}</table>")
    }
}
